php-checks/src/main/java/org/sonar/php/checks/NestedTernaryOperatorsCheck.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.tree.expression.ConditionalExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionExpressionTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = "S3358")
+public class NestedTernaryOperatorsCheck extends PHPVisitorCheck {
+
+  @Override
+  public void visitConditionalExpression(ConditionalExpressionTree tree) {
+    TernaryVisitor visitor = new TernaryVisitor();
+    tree.condition().accept(visitor);
+    ExpressionTree trueExpression = tree.trueExpression();
```

Comment:
There is no need to visit true or false expressions if the issue was detected in the condition.
There are some unecessary visit done when an issue is found. 
As such I would first make the subvisitor visit the false expression then if no issue found, visit the true expression if not null and only then if no issue found, the condition expression.

I am not sure of the choice behind reporting nested on false expression only. Why not provide all nested children as secondaries ? This would avoid reporting again on the true expression after fixing the first issue.

Reply:
I changed to collect from 1 to 3 sub ternary expression (in false, true, condition).